src/representations/calculator_spherical_expansion.hh

```diff
@@ -1046,21 +1044,14 @@ namespace rascal {
     using internal::AtomicSmearingType;
     using internal::RadialBasisType;
 
-    switch (internal::combineEnums(this->radial_integral_type,
-                                   this->atomic_smearing_type)) {
-    case internal::combineEnums(RadialBasisType::GTO,
-                                AtomicSmearingType::Constant): {
+    assert(this->atomic_smearing_type == AtomicSmearingType::Constant);
+    if (this->radial_integral_type == RadialBasisType::GTO) {
```

Comment:
How would you deal with a combination of 3, 4 .... template arguments that can't be all mixed together ? Also why go for if statement instead of switch ?

Reply:
> How would you deal with a combination of 3, 4 .... template arguments that can't be all mixed together ?

```
if (a == TypeA::FOO && b == TypeB::BAR) {
}
```

> Also why go for if statement instead of switch ?

Because they allow to use && and || to combine values. Also, they naturally introduce a new scope, which mean we can define new variables and so on.